src/pytest_memray/marks.py

```diff
@@ -162,15 +184,30 @@ def _passes_filter(
 
 
 def limit_memory(
-    limit: str, *, _result_file: Path, _config: Config
-) -> _MemoryInfo | None:
+    limit: str,
+    *,
+    _result_file: Path,
+    _config: Config,
+    _test_id: str,
+) -> _MemoryInfo | _MoreMemoryInfo | None:
     """Limit memory used by the test."""
     reader = FileReader(_result_file)
     allocations: list[AllocationRecord] = list(
         reader.get_high_watermark_allocation_records(merge_threads=True)
     )
     max_memory = parse_memory_string(limit)
     total_allocated_memory = sum(record.size for record in allocations)
+
+    if _config.cache is not None:
+        cache = _config.cache.get(f"memray/{_test_id}", {})
+        previous = cache.get("total_allocated_memory", float("inf"))
+        fail_on_increase = cast(bool, value_or_ini(_config, "fail_on_increase"))
+        if fail_on_increase and total_allocated_memory > previous:
+            return _MoreMemoryInfo(previous, total_allocated_memory)
+
+        cache["total_allocated_memory"] = total_allocated_memory
+        _config.cache.set(f"memray/{_test_id}", cache)
```

Comment:
The implementation all looks good to me, but I think we should clarify the docs a bit. The cache is only updated if the test succeeds, but I don't think that's implied by the current wording:

> `--fail-on-increase` - Fail if the memory usage increases from previous runs in tests marked with "limit_memory"

I think we should probably instead say something like:

> `--fail-on-increase` - Fail a test with the `limit_memory` marker if it uses more memory than its last successful run

We also probably want to make it clear that this relies on pytest's caching mechanism, and so it only applies between two runs in the same environment. We don't have to say that everywhere - I think we can stick with our one sentence summary in most places - but at least in the `configuration.rst` and possibly in the README.md, I'd suggest mentioning that this only applies to two runs on the same computer which are able to share a cache, as well as possibly walking through an example of how someone is intended to use this new feature (e.g. running a test with the old implementation of an algorithm, switching to a new implementation, and re-running the test with `--

Reply:
Done